### PR TITLE
Feature 247 search skills

### PR DESCRIPTION
**Pull request by gustavo-rojas-jalafund**
[AB#247](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/247) https://dev.azure.com/funjaladev/DevLevel3/_workitems/edit/247/

# Overview

Add button next to Input field. -> "Search" or "Find"
The user will enter a skill name and trigger a call to the API
Once the Search button is clicked, call the api, retrieve the skills and show them to the user

### CODE REVIEW

File: src/components/resumes/Details.jsx [hunk 1]
```diff
@@ -56,6 +80,15 @@ function Details() {
     }
   };
 
+  const getSkills = async (event) => {
+    console.log("Skill input: ", skillInput);
+    event.preventDefault();
+    const response = await getSkillsByName(skillInput);
+    console.log(response);
+  };
```
Comment (ela-ocana-jalafund) [inline src/components/resumes/Details.jsx:87]: you should delete console.log in line 84 and 87

File: src/components/resumes/Details.jsx [hunk 2]
```diff
@@ -56,6 +80,15 @@ function Details() {
...
+
+  const [skillInput, setSkillInput] = useState("");
+
   const handleDelete = () => {};
 
   const initialValues = {
```
Comment (ela-ocana-jalafund) [inline src/components/resumes/Details.jsx:94]: Why are you using initialValues?

File: src/components/resumes/Details.jsx [hunk 3]
```diff
@@ -1,20 +1,37 @@
...
 import Grid from "@material-ui/core/Grid";
+import IconButton from "@material-ui/core/IconButton";
+import InputBase from "@material-ui/core/InputBase";
 import Loading from "./Loading";
+import MenuIcon from "@material-ui/icons/Menu";
 import NotFound from "./NotFound";
 import Paper from "@material-ui/core/Paper";
+import SearchIcon from "@material-ui/icons/Search";
 import TextField from "@material-ui/core/TextField";
 import avatar from "../../assets/img_avatar.jpg";
```
Comment (paola-quintanilla-jalafund) [inline src/components/resumes/Details.jsx:5]: you can group the imports of materialUI something like that:
import {
  Button,
  Container,
  Divider,
  Grid,
  Typography,
} from "@material-ui/core";

File: src/components/resumes/ResumesAPI.js [hunk 4]
```diff
@@ -64,3 +65,25 @@ export async function getResume(id) {
     return { data, handlerError };
   }
 }
+
+export async function getSkillsByName(name) {
+  let handlerError = false;
+  const data = {};
+  try {
+    const { data } = await axios.get(
+      `${API_HOST}/api/${API_VERSION}/${API_NAME_SKILLS}?name=${name}`
```
Comment (paola-quintanilla-jalafund) [inline src/components/resumes/ResumesAPI.js]: can you add a variable for url:
const url = `${API_HOST}/api/${API_VERSION}/${API_NAME_SKILLS}?name=${name}`;
and you use only url;

File: src/components/resumes/Details.jsx [hunk 5]
```diff
@@ -56,6 +77,15 @@ function Details() {
...
 
+  const getSkills = async (event) => {
+    console.log("Skill input: ", skillInput);
+    event.preventDefault();
+    const response = await getSkillsByName(skillInput);
+    console.log(response);
+  };
+
+  const [skillInput, setSkillInput] = useState("");
+
```
Comment (ela-ocana-jalafund) [inline src/components/resumes/Details.jsx]: You con move this useState to group useState just in a place.

File: src/components/resumes/Details.jsx (hunk 3 above)
Comment (gustavo-rojas-jalafund) [reply src/components/resumes/Details.jsx:5]: Done

File: src/components/resumes/ResumesAPI.js (hunk 4 above)
Comment (gustavo-rojas-jalafund) [reply src/components/resumes/ResumesAPI.js]: Done

File: src/components/resumes/Details.jsx (hunk 5 above)
Comment (gustavo-rojas-jalafund) [reply src/components/resumes/Details.jsx]: Done

File: src/components/resumes/Details.jsx (hunk 1 above)
Comment (gustavo-rojas-jalafund) [reply src/components/resumes/Details.jsx:87]: Done

**Review by paola-quintanilla-jalafund — DISMISSED** [review]

**Review by ela-ocana-jalafund — DISMISSED** [review]

File: src/components/resumes/ResumesAPI.js [hunk 6]
```diff
@@ -64,3 +65,23 @@ export async function getResume(id) {
...
 }
+
+export async function getSkillsByName(name) {
+  let handlerError = false;
+  const data = {};
+  try {
+    const { data } = await axios.get(`${API_NAME_SKILLS}?name=${name}`);
+    return { data, handlerError };
+  } catch (error) {
+    // TODO error handling
```
Comment (jorge-lema-jalafund) [inline src/components/resumes/ResumesAPI.js]: remove unnecessary comment

**Review by jorge-lema-jalafund — CHANGES_REQUESTED** [review]

File: src/components/resumes/ResumesAPI.js (hunk 6 above)
Comment (gustavo-rojas-jalafund) [reply src/components/resumes/ResumesAPI.js]: Done

**Review by ela-ocana-jalafund — APPROVED** [review]

**Review by jorge-lema-jalafund — APPROVED** [review]

**Review by paola-quintanilla-jalafund — APPROVED** [review]

**Pull request merged**